index.js

```diff
@@ -1039,6 +1039,68 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  //# takeWhile :: (Foldable f, Alternative f) => (a -> Boolean) -> f a -> f a
+  //.
+  //. Discards the first inner value which does not satisfy the predicate, and
+  //. all subsequent inner values.
+  //.
+  //. ```javascript
+  //. > S.takeWhile(S.odd, [3, 3, 3, 7, 6, 3, 5, 4])
+  //. [3, 3, 3, 7]
+  //.
+  //. > S.takeWhile(S.even, [3, 3, 3, 7, 6, 3, 5, 4])
+  //. []
+  //. ```
+  function Array$takeWhile(pred, xs) {
+    for (var idx = 0; idx < xs.length && pred(xs[idx]); idx += 1);
```

Comment:
The empty block is unnecessary. It can be replaced by a semicolon. But I'll make the change nonetheless.